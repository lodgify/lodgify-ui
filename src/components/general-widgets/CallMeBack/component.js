import React from 'react';
import PropTypes from 'prop-types';

import {
  CALL_ME_BACK,
  DATE,
  EMAIL,
  NAME,
  NOTES,
  PHONE,
  PROPERTY,
  SEND,
  TIME,
} from 'utils/default-strings';
import { Dropdown } from 'inputs/Dropdown';
import { Form } from 'collections/Form';
import { ICON_NAMES } from 'elements/Icon';
import { InputGroup } from 'collections/InputGroup';
import { PhoneInput } from 'inputs/PhoneInput';
import { SingleDatePicker } from 'inputs/SingleDatePicker';
import { TextArea } from 'inputs/TextArea';
import { TextInput } from 'inputs/TextInput';

/**
 * The standard widget for a user to request a call back.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({
  dateInputPlaceholder,
  emailInputLabel,
  errorMessage,
  headingText,
  nameInputLabel,
  notesInputLabel,
  onSubmit,
  phoneInputLabel,
  propertyInputLabel,
  propertyOptions,
  submitButtonText,
  successMessage,
  timeInputLabel,
  timeOptions,
  validation,
}) => (
  <Form
    autoComplete="off"
    errorMessage={errorMessage}
    headingText={headingText}
    onSubmit={onSubmit}
    submitButtonText={submitButtonText}
    successMessage={successMessage}
    validation={validation}
  >
    <InputGroup>
      <TextInput autoComplete="name" label={nameInputLabel} name="name" />
      <PhoneInput autoComplete="tel" label={phoneInputLabel} name="phone" />
    </InputGroup>
    <TextInput autoComplete="email" label={emailInputLabel} name="email" />
    <InputGroup>
      <SingleDatePicker name="date" placeholderText={dateInputPlaceholder} />
      <Dropdown
        icon={ICON_NAMES.CLOCK}
        label={timeInputLabel}
        name="time"
        options={timeOptions}
      />
    </InputGroup>
    <Dropdown
      label={propertyInputLabel}
      name="property"
      options={propertyOptions}
    />
    <TextArea label={notesInputLabel} name="notes" />
  </Form>
);

Component.displayName = 'CallMeBack';

Component.defaultProps = {
  dateInputPlaceholder: DATE,
  emailInputLabel: EMAIL,
  errorMessage: '',
  headingText: CALL_ME_BACK,
  nameInputLabel: NAME,
  notesInputLabel: NOTES,
  onSubmit: Function.prototype,
  phoneInputLabel: PHONE,
  propertyInputLabel: PROPERTY,
  submitButtonText: SEND,
  successMessage: '',
  timeInputLabel: TIME,
  validation: {},
};

Component.propTypes = {
  /** The placeholder for the date input. */
  dateInputPlaceholder: PropTypes.string,
  /** The label for the email input. */
  emailInputLabel: PropTypes.string,
  /** The message to display when the form has an error. */
  errorMessage: PropTypes.string,
  /** The text to display as a heading at the top of the widget. */
  headingText: PropTypes.string,
  /** The label for the name input. */
  nameInputLabel: PropTypes.string,
  /** The label for the notes input. */
  notesInputLabel: PropTypes.string,
  /** The function to call when the form is submitted
   *  @param {Object} values - The values of the inputs in the form.
   */
  onSubmit: PropTypes.func,
  /** The label for the phone input */
  phoneInputLabel: PropTypes.string,
  /** The label for the property input. */
  propertyInputLabel: PropTypes.string,
  /** The options which the user can select for the property field. */
  propertyOptions: PropTypes.arrayOf(
    PropTypes.shape({
      /** The visible text for the option. */
      text: PropTypes.string.isRequired,
      /** The underlying value for the option. */
      value: PropTypes.oneOfType([
        PropTypes.bool,
        PropTypes.number,
        PropTypes.string,
      ]),
    })
  ).isRequired,
  /** The form submit button text */
  submitButtonText: PropTypes.string,
  /** The message to display when the form is successful. */
  successMessage: PropTypes.string,
  /** The label for the time input. */
  timeInputLabel: PropTypes.string,
  /** The options which the user can select for the time field. */
  timeOptions: PropTypes.arrayOf(
    PropTypes.shape({
      /** The visible text for the option. */
      text: PropTypes.string.isRequired,
      /** The underlying value for the option. */
      value: PropTypes.oneOfType([
        PropTypes.bool,
        PropTypes.number,
        PropTypes.string,
      ]),
    })
  ).isRequired,
  /** Settings for validating inputs. Each value should match [the shape documented in `Form`](https://lodgify.github.io/lodgify-ui/#/Collections/Form) */
  validation: PropTypes.shape({
    date: PropTypes.object,
    email: PropTypes.object,
    name: PropTypes.object,
    notes: PropTypes.object,
    phone: PropTypes.object,
    property: PropTypes.object,
    time: PropTypes.object,
  }),
};
