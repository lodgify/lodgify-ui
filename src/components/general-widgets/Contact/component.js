import React from 'react';
import PropTypes from 'prop-types';

import { Form } from 'collections/Form';
import { InputGroup } from 'collections/InputGroup';
import { TextInput } from 'inputs/TextInput';
import { PhoneInput } from 'inputs/PhoneInput';
import { NumberInput } from 'inputs/NumberInput';
import { DateRangePicker } from 'inputs/DateRangePicker';
import { TextArea } from 'inputs/TextArea';
import { CaptchaInput } from 'inputs/CaptchaInput';
import { Dropdown } from 'inputs/Dropdown';
import {
  ARRIVAL,
  CONTACT,
  COMMENTS,
  DEPARTURE,
  EMAIL,
  GUESTS,
  NAME,
  PHONE,
  PROPERTY,
  ROOM,
  SECURITY_CODE,
  SEND,
} from 'utils/default-strings';

/**
 * The standard widget for a user contact an owner.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({
  arrivalDateInputLabel,
  captchaInputImage,
  captchaInputLabel,
  commentsInputLabel,
  dateRangePickerLocaleCode,
  departureDateInputLabel,
  emailInputLabel,
  errorMessage,
  guestsInputLabel,
  headingText,
  nameInputLabel,
  onChangeProperty,
  onSubmit,
  phoneInputLabel,
  propertyInputLabel,
  propertyOptions,
  roomInputLabel,
  roomOptions,
  submitButtonText,
  successMessage,
  validation,
}) => (
  <Form
    autoComplete="off"
    errorMessage={errorMessage}
    headingText={headingText}
    onSubmit={onSubmit}
    submitButtonText={submitButtonText}
    successMessage={successMessage}
    validation={validation}
  >
    <InputGroup>
      <TextInput autoComplete="name" label={nameInputLabel} name="name" />
      <PhoneInput autoComplete="tel" label={phoneInputLabel} name="phone" />
    </InputGroup>
    <TextInput autoComplete="email" label={emailInputLabel} name="email" />
    <InputGroup>
      <DateRangePicker
        endDatePlaceholderText={departureDateInputLabel}
        localeCode={dateRangePickerLocaleCode}
        name="dates"
        startDatePlaceholderText={arrivalDateInputLabel}
        width="eight"
      />
      <NumberInput
        label={guestsInputLabel}
        min={1}
        name="guests"
        width="four"
      />
    </InputGroup>
    <TextArea label={commentsInputLabel} name="comments" />
    {(roomOptions || propertyOptions) && (
      <InputGroup>
        {propertyOptions && (
          <Dropdown
            label={propertyInputLabel}
            name="property"
            onChange={onChangeProperty}
            options={propertyOptions}
          />
        )}
        {roomOptions && (
          <Dropdown label={roomInputLabel} name="room" options={roomOptions} />
        )}
      </InputGroup>
    )}
    <CaptchaInput
      image={captchaInputImage}
      label={captchaInputLabel}
      name="captcha"
    />
  </Form>
);

Component.displayName = 'Contact';

Component.defaultProps = {
  arrivalDateInputLabel: ARRIVAL,
  captchaInputLabel: SECURITY_CODE,
  commentsInputLabel: COMMENTS,
  dateRangePickerLocaleCode: undefined,
  departureDateInputLabel: DEPARTURE,
  emailInputLabel: EMAIL,
  errorMessage: '',
  guestsInputLabel: GUESTS,
  headingText: CONTACT,
  nameInputLabel: NAME,
  onChangeProperty: Function.prototype,
  onSubmit: Function.prototype,
  phoneInputLabel: PHONE,
  propertyInputLabel: PROPERTY,
  propertyOptions: null,
  roomInputLabel: ROOM,
  roomOptions: null,
  submitButtonText: SEND,
  successMessage: '',
  validation: {},
};

Component.propTypes = {
  /** The label for the arrival date input.*/
  arrivalDateInputLabel: PropTypes.string,
  /** The source url for the image to display. */
  captchaInputImage: PropTypes.string.isRequired,
  /** The label for the captcha input.*/
  captchaInputLabel: PropTypes.string,
  /** The label for the comments input.*/
  commentsInputLabel: PropTypes.string,
  /** The ISO 639-1 locale code which changes the format and language of days of the week and the months of the year in the date range picker. */
  dateRangePickerLocaleCode: PropTypes.string,
  /** The label for the departure date input.*/
  departureDateInputLabel: PropTypes.string,
  /** The label for the email input.*/
  emailInputLabel: PropTypes.string,
  /** The message to display when the form has an error. */
  errorMessage: PropTypes.string,
  /** The label for the guests input.*/
  guestsInputLabel: PropTypes.string,
  /** The text to display as a heading at the top of the widget. */
  headingText: PropTypes.string,
  /** The label for the name input.*/
  nameInputLabel: PropTypes.string,
  /** The function called when the property dropdown is changed.
   *  @param {String}        name - The name of the property dropdown field.
   *  @param {String|Number} value - The value of the property dropdown after the change
   */
  onChangeProperty: PropTypes.func,
  /** The function to call when the form is submitted
   *  @param {Object} values - The values of the inputs in the form.
   */
  onSubmit: PropTypes.func,
  /** The label for the phone input.*/
  phoneInputLabel: PropTypes.string,
  /** The label for the property input.*/
  propertyInputLabel: PropTypes.string,
  /** The options which the user can select for the property field. */
  propertyOptions: PropTypes.arrayOf(
    PropTypes.shape({
      /** The indent level of a property option. One of: 0, 1, 2, 3, 4, 5 */
      indent: PropTypes.oneOf([0, 1, 2, 3, 4, 5]),
      /** The visible text for the option. */
      text: PropTypes.string.isRequired,
      /** The underlying value for the option. */
      value: PropTypes.oneOfType([
        PropTypes.bool,
        PropTypes.number,
        PropTypes.string,
      ]),
    })
  ),
  /** The label for the room input.*/
  roomInputLabel: PropTypes.string,
  /** The options which the user can select for the room field. */
  roomOptions: PropTypes.arrayOf(
    PropTypes.shape({
      /** The visible text for the option. */
      text: PropTypes.string.isRequired,
      /** The underlying value for the option. */
      value: PropTypes.oneOfType([
        PropTypes.bool,
        PropTypes.number,
        PropTypes.string,
      ]),
    })
  ),
  /** The text to display on the submit button. */
  submitButtonText: PropTypes.string,
  /** The message to display when the form is successful. */
  successMessage: PropTypes.string,
  /** Settings for validating inputs. Each value should match [the shape documented in `Form`](https://lodgify.github.io/lodgify-ui/#/Collections/Form) */
  validation: PropTypes.shape({
    captcha: PropTypes.object,
    comments: PropTypes.object,
    dates: PropTypes.object,
    email: PropTypes.object,
    guests: PropTypes.object,
    name: PropTypes.object,
    phone: PropTypes.object,
    property: PropTypes.object,
    room: PropTypes.object,
  }),
};
