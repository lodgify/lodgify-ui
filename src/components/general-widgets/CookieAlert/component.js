import React from 'react';
import PropTypes from 'prop-types';

import { Modal } from 'elements/Modal';
import { FlexContainer } from 'layout/FlexContainer';
import { Button } from 'elements/Button';
import { Divider } from 'elements/Divider';
import { HorizontalGutters } from 'layout/HorizontalGutters';
import { Paragraph } from 'typography/Paragraph';
import { withResponsive } from 'utils/with-responsive';
import { Link } from 'elements/Link';

import { getFormMarkup } from './utils/getFormMarkup';

const Component = ({
  buttonText,
  isOpen,
  isUserOnMobile,
  linkText,
  linkUrl,
  onAccept,
  text,
}) => {
  if (!isOpen) return false;
  return isUserOnMobile ? (
    <Modal hasCloseIcon={false} hasRoundedCorners isOpen>
      <HorizontalGutters>
        <FlexContainer
          alignItems="center"
          flexDirection="column"
          justifyContent="center"
        >
          <Divider />
          <Paragraph>{text}</Paragraph>
          <Link href={linkUrl}>{linkText}</Link>
          <Divider />
          <Button onClick={onAccept}>{buttonText}</Button> <Divider />
        </FlexContainer>
      </HorizontalGutters>
    </Modal>
  ) : (
    <div className="cookies-form">
      {getFormMarkup(buttonText, linkText, linkUrl, onAccept, text)}
    </div>
  );
};

Component.displayName = 'CookieAlert';

Component.defaultProps = {
  buttonText: 'Accept',
  isOpen: false,
  linkText: 'Learn More',
  linkUrl: '/',
};

Component.propTypes = {
  /** The text to display on the button inside the component. */
  buttonText: PropTypes.string,
  /** Is the modal open. */
  isOpen: PropTypes.bool,
  /**
   * Is the user on a mobile device.
   * Provided by `withResponsive` so ignored in the styleguide.
   * @ignore
   */
  isUserOnMobile: PropTypes.bool.isRequired,
  /** The text to display as a link. */
  linkText: PropTypes.string,
  /** The url attached to the link. */
  linkUrl: PropTypes.string,
  /** The on accept handler for the call to action. */
  onAccept: PropTypes.func.isRequired,
  /** The text to display in the component. */
  text: PropTypes.string.isRequired,
  /** The element to be clicked to display the modal. */
};

export const ComponentWithResponsive = withResponsive(Component);
