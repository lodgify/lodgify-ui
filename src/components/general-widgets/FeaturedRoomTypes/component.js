import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

import { Heading } from 'typography/Heading';
import { FeaturedRoomType } from 'general-widgets/FeaturedRoomType';
import { buildKeyFromStrings } from 'utils/build-key-from-strings';

import { PLACEHOLDERS } from './constants';

/**
 * The standard widget for displaying a list of featured rooms.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({
  featuredRoomTypes,
  headingText,
  isShowingPlaceholder,
}) => {
  const featuredRoomTypesToMap =
    isShowingPlaceholder && featuredRoomTypes.length === 0
      ? PLACEHOLDERS
      : featuredRoomTypes;

  return (
    <Fragment>
      {headingText && <Heading>{headingText}</Heading>}
      {featuredRoomTypesToMap.map((featuredRoomType, index) => (
        <FeaturedRoomType
          isShowingPlaceholder={isShowingPlaceholder}
          key={buildKeyFromStrings(featuredRoomType.roomTypeName, index)}
          {...featuredRoomType}
        />
      ))}
    </Fragment>
  );
};

Component.displayName = 'FeaturedRoomTypes';

Component.defaultProps = {
  headingText: null,
  isShowingPlaceholder: false,
  featuredRoomTypes: [],
};

Component.propTypes = {
  /** An array of [`FeaturedRoomType`](#/FeaturedRoomType) props objects. */
  featuredRoomTypes: PropTypes.arrayOf(PropTypes.object),
  /** The text to display as a heading at the top of the widget. */
  headingText: PropTypes.string,
  /** Is the component showing placeholders to reserve space for content which will appear. */
  isShowingPlaceholder: PropTypes.bool,
};
