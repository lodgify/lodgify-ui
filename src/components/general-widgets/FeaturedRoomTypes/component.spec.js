import React from 'react';
import { mount } from 'enzyme';
import { expectComponentToHaveDisplayName } from '@lodgify/enzyme-jest-expect-helpers';

import { Component as FeaturedRoomTypes } from './component';

const featuredRoomTypes = [
  {
    bedsNumber: 3,
    guestsNumber: 3,
    imageUrl:
      'https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg',
    locationName: 'Catania',
    nightPrice: '$280',
    ratingNumber: 4.8,
    roomTypeName: 'The Cat House',
    roomTypeUrl: '/',
  },
  {
    bedsNumber: 3,
    guestsNumber: 3,
    imageUrl:
      'https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg',
    locationName: 'Catania',
    nightPrice: '$280',
    ratingNumber: 4.8,
    roomTypeName: 'The Cat House',
    roomTypeUrl: '/',
  },
];

const getFeaturedRoomTypes = otherProps =>
  mount(
    <FeaturedRoomTypes featuredRoomTypes={featuredRoomTypes} {...otherProps} />
  );

describe('<FeaturedRoomTypes />', () => {
  describe('by default', () => {
    it('should render the right structure', () => {
      const actual = getFeaturedRoomTypes();

      expect(actual).toMatchSnapshot();
    });
  });

  describe('if `props.headingText` is passed', () => {
    it('should render the right structure', () => {
      const headingText = 'My god I love Elton John';
      const actual = getFeaturedRoomTypes({ headingText });

      expect(actual).toMatchSnapshot();
    });
  });

  it('should have `displayName` `FeaturedRoomTypes`', () => {
    expectComponentToHaveDisplayName(FeaturedRoomTypes, 'FeaturedRoomTypes');
  });
});
