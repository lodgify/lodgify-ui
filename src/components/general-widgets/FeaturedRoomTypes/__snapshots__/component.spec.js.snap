// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeaturedRoomTypes /> by default should render the right structure 1`] = `
<FeaturedRoomTypes
  featuredRoomTypes={
    Array [
      Object {
        "bedsNumber": 3,
        "guestsNumber": 3,
        "imageUrl": "https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg",
        "locationName": "Catania",
        "nightPrice": "$280",
        "ratingNumber": 4.8,
        "roomTypeName": "The Cat House",
        "roomTypeUrl": "/",
      },
      Object {
        "bedsNumber": 3,
        "guestsNumber": 3,
        "imageUrl": "https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg",
        "locationName": "Catania",
        "nightPrice": "$280",
        "ratingNumber": 4.8,
        "roomTypeName": "The Cat House",
        "roomTypeUrl": "/",
      },
    ]
  }
  headingText={null}
>
  <Grid
    areColumnsCentered={false}
  >
    <Grid
      centered={false}
    >
      <div
        className="ui grid"
      >
        <GridColumn
          computer={4}
          key="The Cat House0"
          mobile={12}
          tablet={6}
          verticalAlignContent="top"
          width={null}
        >
          <GridColumn
            computer={4}
            mobile={12}
            tablet={6}
            verticalAlign="top"
            width={null}
          >
            <div
              className="top aligned four wide computer twelve wide mobile six wide tablet column"
            >
              <FeaturedRoomType
                bedsLabel="Beds"
                bedsNumber={3}
                guestsLabel="Guests"
                guestsNumber={3}
                imageUrl="https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg"
                locationName="Catania"
                nightPrice="$280"
                ratingNumber={4.8}
                roomTypeName="The Cat House"
                roomTypeUrl="/"
              >
                <Card
                  className="has-featured"
                  href="/"
                >
                  <a
                    className="ui card has-featured"
                    href="/"
                    onClick={[Function]}
                  >
                    <ResponsiveImage
                      alternativeText="Image Widget"
                      hasRoundedCorners={false}
                      imageHeight={null}
                      imageNotFoundLabelText="Image not found!"
                      imageTitle="Image Title"
                      imageUrl="https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg"
                      imageWidth={null}
                      isAvatar={false}
                      isCircular={false}
                      isFluid={true}
                      label={null}
                      sizes={null}
                      srcSet={null}
                    >
                      <figure
                        className="responsive-image is-fluid"
                      >
                        <Image
                          alt="Image Widget"
                          as="img"
                          avatar={false}
                          fluid={true}
                          onLoad={[Function]}
                          sizes={null}
                          src="https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg"
                          srcSet={null}
                          title="Image Title"
                          ui={true}
                        >
                          <img
                            alt="Image Widget"
                            className="ui fluid image"
                            onLoad={[Function]}
                            sizes={null}
                            src="https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg"
                            srcSet={null}
                            title="Image Title"
                          />
                        </Image>
                      </figure>
                    </ResponsiveImage>
                    <CardContent>
                      <div
                        className="content"
                      >
                        <CardHeader>
                          <div
                            className="header"
                          >
                            The Cat House
                          </div>
                        </CardHeader>
                        <CardDescription>
                          <div
                            className="description"
                          >
                            Catania
                          </div>
                        </CardDescription>
                        <CardDescription>
                          <div
                            className="description"
                          >
                            Guests: 3 | Beds: 3
                          </div>
                        </CardDescription>
                        <CardDescription>
                          <div
                            className="description"
                          >
                            <span>
                              from 
                              <Heading
                                isColorInverted={false}
                                size="medium"
                                textAlign={null}
                              >
                                <Header
                                  as="h3"
                                  inverted={false}
                                  textAlign={null}
                                >
                                  <h3
                                    className="ui header"
                                  >
                                    $280
                                  </h3>
                                </Header>
                              </Heading>
                              /night
                            </span>
                          </div>
                        </CardDescription>
                      </div>
                    </CardContent>
                  </a>
                </Card>
              </FeaturedRoomType>
            </div>
          </GridColumn>
        </GridColumn>
        <GridColumn
          computer={4}
          key="The Cat House1"
          mobile={12}
          tablet={6}
          verticalAlignContent="top"
          width={null}
        >
          <GridColumn
            computer={4}
            mobile={12}
            tablet={6}
            verticalAlign="top"
            width={null}
          >
            <div
              className="top aligned four wide computer twelve wide mobile six wide tablet column"
            >
              <FeaturedRoomType
                bedsLabel="Beds"
                bedsNumber={3}
                guestsLabel="Guests"
                guestsNumber={3}
                imageUrl="https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg"
                locationName="Catania"
                nightPrice="$280"
                ratingNumber={4.8}
                roomTypeName="The Cat House"
                roomTypeUrl="/"
              >
                <Card
                  className="has-featured"
                  href="/"
                >
                  <a
                    className="ui card has-featured"
                    href="/"
                    onClick={[Function]}
                  >
                    <ResponsiveImage
                      alternativeText="Image Widget"
                      hasRoundedCorners={false}
                      imageHeight={null}
                      imageNotFoundLabelText="Image not found!"
                      imageTitle="Image Title"
                      imageUrl="https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg"
                      imageWidth={null}
                      isAvatar={false}
                      isCircular={false}
                      isFluid={true}
                      label={null}
                      sizes={null}
                      srcSet={null}
                    >
                      <figure
                        className="responsive-image is-fluid"
                      >
                        <Image
                          alt="Image Widget"
                          as="img"
                          avatar={false}
                          fluid={true}
                          onLoad={[Function]}
                          sizes={null}
                          src="https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg"
                          srcSet={null}
                          title="Image Title"
                          ui={true}
                        >
                          <img
                            alt="Image Widget"
                            className="ui fluid image"
                            onLoad={[Function]}
                            sizes={null}
                            src="https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg"
                            srcSet={null}
                            title="Image Title"
                          />
                        </Image>
                      </figure>
                    </ResponsiveImage>
                    <CardContent>
                      <div
                        className="content"
                      >
                        <CardHeader>
                          <div
                            className="header"
                          >
                            The Cat House
                          </div>
                        </CardHeader>
                        <CardDescription>
                          <div
                            className="description"
                          >
                            Catania
                          </div>
                        </CardDescription>
                        <CardDescription>
                          <div
                            className="description"
                          >
                            Guests: 3 | Beds: 3
                          </div>
                        </CardDescription>
                        <CardDescription>
                          <div
                            className="description"
                          >
                            <span>
                              from 
                              <Heading
                                isColorInverted={false}
                                size="medium"
                                textAlign={null}
                              >
                                <Header
                                  as="h3"
                                  inverted={false}
                                  textAlign={null}
                                >
                                  <h3
                                    className="ui header"
                                  >
                                    $280
                                  </h3>
                                </Header>
                              </Heading>
                              /night
                            </span>
                          </div>
                        </CardDescription>
                      </div>
                    </CardContent>
                  </a>
                </Card>
              </FeaturedRoomType>
            </div>
          </GridColumn>
        </GridColumn>
      </div>
    </Grid>
  </Grid>
</FeaturedRoomTypes>
`;

exports[`<FeaturedRoomTypes /> if \`props.headingText\` is passed should render the right structure 1`] = `
<FeaturedRoomTypes
  featuredRoomTypes={
    Array [
      Object {
        "bedsNumber": 3,
        "guestsNumber": 3,
        "imageUrl": "https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg",
        "locationName": "Catania",
        "nightPrice": "$280",
        "ratingNumber": 4.8,
        "roomTypeName": "The Cat House",
        "roomTypeUrl": "/",
      },
      Object {
        "bedsNumber": 3,
        "guestsNumber": 3,
        "imageUrl": "https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg",
        "locationName": "Catania",
        "nightPrice": "$280",
        "ratingNumber": 4.8,
        "roomTypeName": "The Cat House",
        "roomTypeUrl": "/",
      },
    ]
  }
  headingText="My god I love Elton John"
>
  <Grid
    areColumnsCentered={false}
  >
    <Grid
      centered={false}
    >
      <div
        className="ui grid"
      >
        <GridColumn
          verticalAlignContent="top"
          width={12}
        >
          <GridColumn
            verticalAlign="top"
            width={12}
          >
            <div
              className="top aligned twelve wide column"
            >
              <Heading
                isColorInverted={false}
                size="medium"
                textAlign={null}
              >
                <Header
                  as="h3"
                  inverted={false}
                  textAlign={null}
                >
                  <h3
                    className="ui header"
                  >
                    My god I love Elton John
                  </h3>
                </Header>
              </Heading>
            </div>
          </GridColumn>
        </GridColumn>
        <GridColumn
          computer={4}
          key="The Cat House0"
          mobile={12}
          tablet={6}
          verticalAlignContent="top"
          width={null}
        >
          <GridColumn
            computer={4}
            mobile={12}
            tablet={6}
            verticalAlign="top"
            width={null}
          >
            <div
              className="top aligned four wide computer twelve wide mobile six wide tablet column"
            >
              <FeaturedRoomType
                bedsLabel="Beds"
                bedsNumber={3}
                guestsLabel="Guests"
                guestsNumber={3}
                imageUrl="https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg"
                locationName="Catania"
                nightPrice="$280"
                ratingNumber={4.8}
                roomTypeName="The Cat House"
                roomTypeUrl="/"
              >
                <Card
                  className="has-featured"
                  href="/"
                >
                  <a
                    className="ui card has-featured"
                    href="/"
                    onClick={[Function]}
                  >
                    <ResponsiveImage
                      alternativeText="Image Widget"
                      hasRoundedCorners={false}
                      imageHeight={null}
                      imageNotFoundLabelText="Image not found!"
                      imageTitle="Image Title"
                      imageUrl="https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg"
                      imageWidth={null}
                      isAvatar={false}
                      isCircular={false}
                      isFluid={true}
                      label={null}
                      sizes={null}
                      srcSet={null}
                    >
                      <figure
                        className="responsive-image is-fluid"
                      >
                        <Image
                          alt="Image Widget"
                          as="img"
                          avatar={false}
                          fluid={true}
                          onLoad={[Function]}
                          sizes={null}
                          src="https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg"
                          srcSet={null}
                          title="Image Title"
                          ui={true}
                        >
                          <img
                            alt="Image Widget"
                            className="ui fluid image"
                            onLoad={[Function]}
                            sizes={null}
                            src="https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg"
                            srcSet={null}
                            title="Image Title"
                          />
                        </Image>
                      </figure>
                    </ResponsiveImage>
                    <CardContent>
                      <div
                        className="content"
                      >
                        <CardHeader>
                          <div
                            className="header"
                          >
                            The Cat House
                          </div>
                        </CardHeader>
                        <CardDescription>
                          <div
                            className="description"
                          >
                            Catania
                          </div>
                        </CardDescription>
                        <CardDescription>
                          <div
                            className="description"
                          >
                            Guests: 3 | Beds: 3
                          </div>
                        </CardDescription>
                        <CardDescription>
                          <div
                            className="description"
                          >
                            <span>
                              from 
                              <Heading
                                isColorInverted={false}
                                size="medium"
                                textAlign={null}
                              >
                                <Header
                                  as="h3"
                                  inverted={false}
                                  textAlign={null}
                                >
                                  <h3
                                    className="ui header"
                                  >
                                    $280
                                  </h3>
                                </Header>
                              </Heading>
                              /night
                            </span>
                          </div>
                        </CardDescription>
                      </div>
                    </CardContent>
                  </a>
                </Card>
              </FeaturedRoomType>
            </div>
          </GridColumn>
        </GridColumn>
        <GridColumn
          computer={4}
          key="The Cat House1"
          mobile={12}
          tablet={6}
          verticalAlignContent="top"
          width={null}
        >
          <GridColumn
            computer={4}
            mobile={12}
            tablet={6}
            verticalAlign="top"
            width={null}
          >
            <div
              className="top aligned four wide computer twelve wide mobile six wide tablet column"
            >
              <FeaturedRoomType
                bedsLabel="Beds"
                bedsNumber={3}
                guestsLabel="Guests"
                guestsNumber={3}
                imageUrl="https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg"
                locationName="Catania"
                nightPrice="$280"
                ratingNumber={4.8}
                roomTypeName="The Cat House"
                roomTypeUrl="/"
              >
                <Card
                  className="has-featured"
                  href="/"
                >
                  <a
                    className="ui card has-featured"
                    href="/"
                    onClick={[Function]}
                  >
                    <ResponsiveImage
                      alternativeText="Image Widget"
                      hasRoundedCorners={false}
                      imageHeight={null}
                      imageNotFoundLabelText="Image not found!"
                      imageTitle="Image Title"
                      imageUrl="https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg"
                      imageWidth={null}
                      isAvatar={false}
                      isCircular={false}
                      isFluid={true}
                      label={null}
                      sizes={null}
                      srcSet={null}
                    >
                      <figure
                        className="responsive-image is-fluid"
                      >
                        <Image
                          alt="Image Widget"
                          as="img"
                          avatar={false}
                          fluid={true}
                          onLoad={[Function]}
                          sizes={null}
                          src="https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg"
                          srcSet={null}
                          title="Image Title"
                          ui={true}
                        >
                          <img
                            alt="Image Widget"
                            className="ui fluid image"
                            onLoad={[Function]}
                            sizes={null}
                            src="https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg"
                            srcSet={null}
                            title="Image Title"
                          />
                        </Image>
                      </figure>
                    </ResponsiveImage>
                    <CardContent>
                      <div
                        className="content"
                      >
                        <CardHeader>
                          <div
                            className="header"
                          >
                            The Cat House
                          </div>
                        </CardHeader>
                        <CardDescription>
                          <div
                            className="description"
                          >
                            Catania
                          </div>
                        </CardDescription>
                        <CardDescription>
                          <div
                            className="description"
                          >
                            Guests: 3 | Beds: 3
                          </div>
                        </CardDescription>
                        <CardDescription>
                          <div
                            className="description"
                          >
                            <span>
                              from 
                              <Heading
                                isColorInverted={false}
                                size="medium"
                                textAlign={null}
                              >
                                <Header
                                  as="h3"
                                  inverted={false}
                                  textAlign={null}
                                >
                                  <h3
                                    className="ui header"
                                  >
                                    $280
                                  </h3>
                                </Header>
                              </Heading>
                              /night
                            </span>
                          </div>
                        </CardDescription>
                      </div>
                    </CardContent>
                  </a>
                </Card>
              </FeaturedRoomType>
            </div>
          </GridColumn>
        </GridColumn>
      </div>
    </Grid>
  </Grid>
</FeaturedRoomTypes>
`;
